apps/emqx_resource/src/emqx_resource_sup.erl

```diff
@@ -45,7 +45,12 @@ init([]) ->
               restart => transient,
               shutdown => 5000, type => worker, modules => [Mod]}
         end || Idx <- lists:seq(1, ?POOL_SIZE)],
-    {ok, {SupFlags, [Metrics | ResourceInsts]}}.
+    HealthCheck = 
+            #{id => emqx_resource_health_check_sup,
+              start => {emqx_resource_health_check_sup, start_link, []},
+              restart => transient,
+              shutdown => 5000, type => supervisor, modules => [emqx_resource_health_check_sup]},
```

Comment:
If the child process is another supervisor, the `shutdown` time must be set to infinity to give the subtree ample time to shut down.

Reply:
I'll merge this PR now and then change it to `infinity` in another PR.